core/src/main/scala/kafka/log/LogCleanerManager.scala

```diff
@@ -549,7 +550,7 @@ private[log] class LogCleanerManager(val logDirs: Seq[File],
 
   def removeMetrics(): Unit = {
     GaugeMetricNameNoTag.foreach(metricsGroup.removeMetric)
-    GaugeMetricNameWithTag.asScala.foreach(metricNameAndTags => {
+    gaugeMetricNameWithTag.asScala.foreach(metricNameAndTags => {
```

Comment:
we want to remove entries from `gaugeMetricNameWithTag` here else it may be continuously growing

Reply:
nit: `foreach { metricNameAndTags =>`.

Reply:
I am so sorry to introduce the problem. Thanks, @dajac @divijvaidya I have updated.